projects/packages/jetpack-mu-wpcom/src/features/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-launchpad.php

```diff
@@ -139,19 +139,23 @@ public function update_site_options( $request ) {
 		foreach ( $input as $key => $value ) {
 			switch ( $key ) {
 				case 'checklist_statuses':
-					$launchpad_checklist_tasks_statuses_option = (array) get_option( 'launchpad_checklist_tasks_statuses', array() );
-					$launchpad_checklist_tasks_statuses_option = array_merge( $launchpad_checklist_tasks_statuses_option, $value );
-
-					foreach ( $value as $task => $task_value ) {
-						if ( $task_value ) {
-							// If we're marking a task as complete, the value should be truthy, so fire the Tracks event.
-							wpcom_launchpad_track_completed_task( $task );
+					$complete_values   = array();
+					$incomplete_values = array();
+
+					foreach ( $value as $task_id => $task_status ) {
+						if ( $task_status ) {
+							$complete_values[] = $task_id;
+						} else {
+							$incomplete_values[] = $task_id;
 						}
 					}
 
-					if ( update_option( 'launchpad_checklist_tasks_statuses', $launchpad_checklist_tasks_statuses_option ) ) {
+					$updated_complete_tasks   = wpcom_mark_launchpad_task_complete( $complete_values );
+					$updated_incomplete_tasks = wpcom_mark_launchpad_task_incomplete( $incomplete_values );
+					if ( $updated_complete_tasks || $updated_incomplete_tasks ) {
 						$updated[ $key ] = $value;
 					}
```

Comment:
Thought: do we want to return the specific items that were updated? By simply returning `$value`, we may be giving an incorrect picture of what actions were actually taken. This isn't necessarily blocking, but our additional validation means that the return values may not match the changes we're making.

Reply:
This is also related to the outcome of [this other conversation](https://github.com/Automattic/jetpack/pull/31471#discussion_r1285871674)

Reply:
My refactors should take care of this issue as well.